hot-reload-devtools/src/main/kotlin/org/jetbrains/compose/devtools/states/NotificationsUIState.kt

```diff
@@ -0,0 +1,157 @@
+
+
+fun CoroutineScope.launchNotificationsUIState() = launchState(NotificationsUIState) {
+    fun createNotification(
+        type: UINotificationType,
+        title: String,
+        message: String,
+        isDisposableFromUI: Boolean,
+        details: List<String> = emptyList()
+    ) = UINotificationImpl(type, title, message, isDisposableFromUI, details) { notification ->
+        this@launchNotificationsUIState.launch {
+            NotificationsUIState.update { it - notification }
+        }
+    }
+
+    fun UIErrorDescription.toNotification(): UINotification = createNotification(
+        type = UINotificationType.ERROR,
+        title = title,
+        message = message.orEmpty(),
+        isDisposableFromUI = false,
+        details = stacktrace.map { it.toString() }
+    )
+
+    suspend fun checkForJBR() {
+        val javaReleaseFile = JavaHome.current().readReleaseFile()
+        if (javaReleaseFile.implementor.orEmpty().contains("JetBrains", ignoreCase = true)) return
+
+        val warning = createNotification(
+            type = UINotificationType.WARNING,
+            title = "Not running on 'JetBrains Runtime'",
+            message = "You're not running on the JetBrains Runtime. Some of the compose hot reload functionality may not be available.",
+            isDisposableFromUI = true,
+            details = javaReleaseFile.values.map { "${it.key} = ${it.value}"},
+        )
+        NotificationsUIState.update { it + warning }
+    }
+
+    suspend fun collectUIErrors() {
+        val notifiedErrors = mutableMapOf<WindowId, Pair<UIErrorDescription, UINotification>>()
+
+        ErrorUIState.collect { errors ->
+            for ((windowId, error) in errors.errors) {
+                if (windowId !in notifiedErrors) {
+                    val notification = error.toNotification()
+                    notifiedErrors[windowId] = error to notification
+                    NotificationsUIState.update { it + notification }
+                }
+            }
+
+            for (windowId in notifiedErrors.keys - errors.errors.keys) {
+                notifiedErrors.remove(windowId)?.second?.dispose()
+            }
+        }
+    }
+
+    suspend fun collectReloadErrors() {
+        var currentStateNotification: UINotification? = null
+
+        ReloadUIState.collect { state ->
+            if (state is ReloadUIState.Failed) {
+                currentStateNotification?.dispose()
+                currentStateNotification = createNotification(
+                    type =UINotificationType.ERROR,
+                    title = "Reload failed",
+                    message = "Reason: ${state.reason}",
+                    isDisposableFromUI = false,
+                    details = state.logs.map { it.message },
+                )
+                NotificationsUIState.update { it + currentStateNotification }
+            } else {
+                currentStateNotification?.dispose()
+            }
+        }
+    }
+
+    checkForJBR()
```

Comment:
I wonder why you have decided to 'block' the coroutine here for checking the JBR?